pkg/controllers/internalmembercluster/v1beta1/member_controller_test.go

```diff
@@ -1178,3 +1185,296 @@ func TestUpdateResourceStats(t *testing.T) {
 		})
 	}
 }
+
+// failedToStartProvider is a property provider that are expected to fail upon startup.
+type failedToStartProvider struct{}
+
+var _ propertyprovider.PropertyProvider = &failedToStartProvider{}
+
+func (p *failedToStartProvider) Start(_ context.Context, _ *rest.Config) error {
+	return fmt.Errorf("expected to fail")
+}
+
+func (p *failedToStartProvider) Collect(_ context.Context) propertyprovider.PropertyCollectionResponse {
+	return propertyprovider.PropertyCollectionResponse{}
+}
```

Comment:
why not move this into the test that uses this?

Reply:
Hi Ryan! I am not sure if I understand this 100%. This provider is used right in the following test; would you like me to move this to a separate file?